src/bit.rs

```diff
@@ -0,0 +1,169 @@
+/// Yields the bits of a `u64`.
+pub struct BitIterator {
+    /// The value that is being iterated bit-wise
+    value: u64,
+    /// True bits in the `mask` are the bits that have been visited.
+    mask: u64,
+}
+
+impl BitIterator {
+    pub fn new(value: u64) -> BitIterator {
+        BitIterator { value, mask: 0 }
+    }
+
+    /// An efficient skip implementation.
+    ///
+    /// Note: The compiler is smart enough to translate a `skip(n)` into a single shift instruction
+    /// if the code is inlined, however inlining does not always happen.
+    pub fn skip_front(mut self, n: u32) -> Self {
```

Comment:
Should we add `inline(always)` attribute?

Reply:
I don't think this is needed, the note was to explain the motivation of `skip_front`/`skip_back` since the iterator has [`skip`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip) method, and that one is not always inlined.